### DISCUSSION TITLE
useState在IDE里面总数报错！或找不到！

### DISCUSSION THREAD

**Discussion by danatechgithub4** [junerver/ComposeHooks#14] [Ideas]
![image](https://github.com/junerver/ComposeHooks/assets/127659297/5af42761-ae5b-484c-9b34-a9313e7b12b7)

![image](https://github.com/junerver/ComposeHooks/assets/127659297/e7f0b300-041e-4947-942a-84098adc1366)

推荐把
`fun <T> useState(default: T & Any): MutableState<T> {`
改成
`fun <T : Any> useState(default: T): MutableState<T> {`

**Comment by junerver** [comment]
`T & Any`是kt的强制非空泛型，他可以保障`useState`的空安全，因为数值状态要求非空，如果你需要一个可空的state ，你应该使用可空版本 `_useState`

如果你要使用类似 react 的解构申明，请使用 `useGetState`，它返回一个三元组。

**Reply by danatechgithub4** [reply]
我不需要可空的state，改成
`fun <T : Any> useState(default: T): MutableState<T> {`

IED依然支持传null的时候报错。

你可能没理解我报错的意思。等下次我的IDE报红的时候我截张图你就知道了。

**Reply by danatechgithub4** [reply]
我重启了几遍IDE都未复现，下次我一点记得截图。。。

**Comment by danatechgithub4** [comment]
![image](https://github.com/junerver/ComposeHooks/assets/127659297/d02cc194-efdb-42f8-b468-5669f025f3ca)
这种情况是稳定复现

**Reply by junerver** [reply]
这个提示和源码关系不大，idea系的IDE经常这样。

**Comment by junerver** [comment]
`T & Any` 这个声明与 Nothing 类型一样，库使用者一般不关注，但是对于库作者很有意义，我不能因为 ide 的跳转定义问题修改它。

**Reply by danatechgithub4** [reply]
![image](https://github.com/junerver/ComposeHooks/assets/127659297/612a77a0-19a9-4a55-9182-a1968ca416bd)

不知道为什么IDE对  `T & Any`不友好，kotlin确实是支持 `T & Any`

**Comment by junerver** [comment]
> ![image](https://private-user-images.githubusercontent.com/127659297/331063434-612a77a0-19a9-4a55-9182-a1968ca416bd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU4MzkzOTMsIm5iZiI6MTcxNTgzOTA5MywicGF0aCI6Ii8xMjc2NTkyOTcvMzMxMDYzNDM0LTYxMmE3N2EwLTE5YTktNGE1NS05MTgyLWExOTY4Y2E0MTZiZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNlQwNTU4MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OGY5NGY3MDI1MjIzYjU5YzJmMzcxOTZiMmY4M2JmNThkNTlkOGZhMzQzMjA2Mjg1MTA3YzhiMzUyMmVhZmUxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Vyr-S4p7y52YR336fTRWwgdDsLatMFnMoHpLT5EyeWs)
> 
> 不知道为什么IDE对 `T & Any`不友好，kotlin确实是支持 `T & Any`

idea 系的 IDE 日常抽风罢了，比如你看：
![image](https://github.com/junerver/ComposeHooks/assets/13448219/81678d00-b011-4270-bd85-fcca395e07c1)

**Comment by danatechgithub4** [comment]
@junerver 你这个点了方法名，进不去看源代码，太招人骂了，肯定要解决一下。

**Reply by junerver** [reply]
事实上这是ide的问题，与源码有什么关系呢？ide没有正确的建立索引而已
